src/storage/store/account/onchain_event_store.rs

```diff
@@ -0,0 +1,63 @@
+use std::sync::Arc;
+
+use prost::Message;
+
+use super::{make_fid_key, Store, StoreEventHandler};
+use crate::core::error::HubError;
+use crate::proto::hub_event::hub_event::Body;
+use crate::proto::hub_event::{HubEvent, HubEventType, MergeOnChainEventBody};
+use crate::proto::onchain_event::OnChainEvent;
+use crate::storage::db::{RocksDB, RocksDbTransactionBatch, RocksdbError};
+use thiserror::Error;
+
+#[derive(Error, Debug)]
+pub enum OnchainEventStorageError {
+    #[error(transparent)]
+    RocksdbError(#[from] RocksdbError),
+
+    #[error(transparent)]
+    HubError(#[from] HubError),
+}
+
+pub fn merge_onchain_event(
+    db: &RocksDB,
+    onchain_event: OnChainEvent,
+) -> Result<(), OnchainEventStorageError> {
+    let primary_key = make_fid_key(onchain_event.fid as u32);
+    db.put(&primary_key, &onchain_event.encode_to_vec())?;
```

Comment:
This needs the OnChainEvent RootPrefix, right?

Reply:
Copied the same primary key prefix we use in hub now.